transport/handler_server.go

```diff
@@ -319,7 +320,16 @@ func (ht *serverHandlerTransport) HandleStreams(startStream func(*Stream)) {
 				s.buf.put(&recvMsg{data: buf[:n]})
 			}
 			if err != nil {
-				s.buf.put(&recvMsg{err: err})
+				if err, ok := err.(http2.StreamError); ok {
+					code := http2ErrConvTab[err.Code]
+					if code == codes.OK {
+						s.buf.put(&recvMsg{err: io.EOF})
+					} else {
+						s.buf.put(&recvMsg{err: StreamErrorf(code, "%v", err)})
+					}
+				} else {
+					s.buf.put(&recvMsg{err: io.EOF})
```

Comment:
The thing you linked to is about `TestRPCTimeout-4` failing, not PingPong.

Or were you trying to debug and found `TestRPCTimeout` similar to `TestPingPong` somehow?